modules/javascript/src/main/webapp/javascript/atmosphere.js

```diff
@@ -1746,7 +1755,7 @@
                 } else {
                     atmosphere.util.log(_request.logLevel, ["Websocket reconnect maximum try reached " + _requestCount]);
                     if (_canLog('warn')) {
-                        atmosphere.util.warn("Websocket error, reason: " + message.reason);
+                        atmosphere.util.warn("Websocket error, reason: " + (message || {}).reason);
```

Comment:
`message` is being accessed here as a global variable, but hasn't been initialized as such. I added a default value to avoid the "NPE", but I expect this log to end with `reason: undefined`, so further revision is necessary.